spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java

```diff
@@ -144,6 +144,9 @@ public void append(InputStream inputStream, String path) throws IOException {
 	@Override
 	public void close() {
 		try {
+			if (this.readingRaw.get()) {
```

Comment:
Here you are what I think on the matter: https://github.com/artembilan/spring-integration/commit/d79079e339b6aa6f39c6fcd7fb61ada52905543a